docs/src/main/asciidoc/transaction.adoc

```diff
@@ -362,6 +362,17 @@ NOTE: The `event` object represents the transaction ID, and defines `toString()`
 TIP: In listener methods, you can access more information about the transaction in progress by accessing the `TransactionManager`,
 which is a CDI bean and can be ``@Inject``ed.
 
+== Configuring transaction log to be stored in a DataSource
+
+The Narayana project has the capability to store the transaction logs into a JDBC Datasource; this should be our recommendation for users needing transaction recovery capabilities, especially when running in volatile containers.
+
+To enable this capability, you need to set `quarkus.transaction-manager.object-store.type` to `jdbc` explicitly. Also, you can specify a datasource name to be used for the transaction log storage by setting `quarkus.transaction-manager.object-store.datasource`. It will use the default datasource configuration if not specified.
```

Comment:
Good catch, we should say not to use XA transactions.

The Narayana JDBCStore implementation disables auto-commit mode and manages the commit/rollback of the connection itself.

I don't know the difference between `quarkus.datasource.jdbc.transactions=enabled` and `quarkus.datasource.jdbc.transactions=disabled` so I cannot say which setting should be used other than telling them not to use `quarkus.datasource.jdbc.transactions=xa`. Is it possible to fail fast if xa is configured?

Reply:
> I don't know the difference between quarkus.datasource.jdbc.transactions=enabled and quarkus.datasource.jdbc.transactions=disabled

Neither do I, hopefully that's documented somewhere in https://quarkus.io/guides/datasource. But transactions are disabled using this property in tests in this PR, so @zhfeng probably knows?

> Is it possible to fail fast if xa is configured?

I suppose you could add that info to the build items you currently rely on (`JdbcDatasourceSomethingBuildItem`?), and add a check in the `NarayanaProcessor`.

Reply:
It's not really documented, [it just says that](https://quarkus.io/guides/datasource#quarkus-agroal_quarkus.datasource.jdbc.transactions) `enabled` means to use regular JDBC transactions.

But if, like you say, the tests still work with the property set to `disabled` then that is the setting we need in the extension or in the documentation.

Reply:
I'mm not sure we can get the value of `quarkus.datasource.jdbc.transactions` at build time?

As far as I know, `ENABLED` means the agroal uses JDBC `Connection` but not `XAConnection` and commit/rollback the JDBC transaction at the end of the transaction boundary.

So as @mmusgrov said
> Narayana JDBCStore implementation disables auto-commit mode and manages the commit/rollback of the connection itself

I think we have to set it to `disabled`.

Reply:
> I'mm not sure we can get the value of `quarkus.datasource.jdbc.transactions` at build time?

It's a build-time property, so yes in theory that's possible: https://github.com/quarkusio/quarkus/blob/e81ba2645039cbf17bcd190ca3ee1a4c0df06e91/extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceJdbcBuildTimeConfig.java#L26-L32

> As far as I know, `ENABLED` means the agroal uses JDBC `Connection` but not `XAConnection` and commit/rollback the JDBC transaction at the end of the transaction boundary.

The relevant code is here: https://github.com/quarkusio/quarkus/blob/9c2d1258967b6cca3d823aef1222a7fcf020453c/extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSources.java#L285-L299

You know how this works better than I do, so I won't try to guess.

I have to wonder, though: if `JDBCStore` manages the commit/rollback of the connection itself, doesn't that mean that the configuration is irrelevant? I.e. even if transactions are enabled, `JDBCStore` will just ignore that?

Note that transactions being enabled in Agroal does not mean transactions are mandated by Agroal. In Hibernate ORM we can use Agroal without transactions with the default configuration and it works perfectly fine.

It's kind of important because, if one must disable transactions on the datasource used in `JDBCStore`, then having the Narayana extension default to the default datasource (which will typically be used for e.g. Hibernate ORM and with transactions enabled) becomes a source of problems rather than something that helps.

So either:

* `JDBCStore` works just fine even with transactions enabled (because it just doesn't use Agroal's transaction support), and we need to adjust the test to remove that `transactions = disabled`
* Or `JDBCStore` cannot work on a datasource with transactions enabled, and we need to both document it, add failsafes, and probably remove the default value for the `JDBCStore` datasource (force users to set it explicitly).

Reply:
OK, I will remove `transactions = disabled` to see if the test works. Thanks @yrodiere !